syntax = "proto3";

package evomorph;

import "src/protos/common.proto";

option go_package = "evomorph/protos";

// Parameters for starting a job.
message StartRequest {
  // Path to XML file with MorpheusML config.
  string model_path = 1;
  // Output directory for simulation data.
  // Default: [TODO]
  string out_dir = 2;
}

// Reply for starting a job.
message StartReply {
  // Status of response.
  StatusType status = 1;
  // Computed ID of the simulation.
  uint32 id = 2;
}

// Parameters for stopping a job.
message StopRequest {
  // ID of stopped job.
  uint32 id = 1;
}

service Engine {
  rpc Start(StartRequest) returns (StartReply);
  rpc Stop(StopRequest) returns (Status);
}
